vision/detect/set_endpoint_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	visionpb "google.golang.org/genproto/googleapis/cloud/vision/v1"
+)
+
+func TestSetEndpoint(t *testing.T) {
+	ctx := context.Background()
+
+	const endpoint = "eu-vision.googleapis.com:443"
+
+	client, err := setEndpoint(ctx, endpoint)
+	if err != nil {
+		t.Fatalf("setEndpoint: %v", err)
+	}
+	defer client.Close()
+
+	image := &visionpb.Image{
+		Source: &visionpb.ImageSource{
+			GcsImageUri: "gs://cloud-samples-data/vision/text/screen.jpg",
+		},
+	}
+	texts, err := client.DetectTexts(ctx, image, nil, 1)
```

Comment:
Does the output of this change based on the endpoint? Is there a more direct way we can test the endpoint was actually set?

Reply:
I don't think the output changes, this is what they're doing in the canonical

Reply:
@nnegrey do you know if there is a more direct way to test if the endpoint was actually used? Or is it completely transparent for users?

Reply:
Not that I know of for this API, you can pass an invalid endpoint and get an error. But no resource is created. 

Natural Language for example has a different set of endpoints they support as documented here: https://cloud.google.com/natural-language/docs/locations

Reply:
Could we remove this portion of the test if it's not testing the sample? Or is it needed to make sure it still works after changing the endpoint?

Reply:
Mostly to just verify the endpoint works.

Reply:
@tbpg how do you feel about this? Would you still want to change anything or should we go ahead and merge?

Reply:
Sounds good to me.

Reply:
In the test, you could set the endpoint to a dummy gRPC server.